tests/infra/health_watcher.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+from enum import Enum, auto
+from collections import Counter
+import time
+import threading
+
+from loguru import logger as LOG
+
+# How often each node in the network is polled
+DEFAULT_POLLING_INTERVAL_S = 1
+
+# Maximum duration after which unestablished connections consider
+# the node as unreachable
+DEFAULT_CLIENT_NODE_TIMEOUT_S = 3
+
+# Maximum number of unsuccessful elections after which the
+# network is considered as _not_ self-healable and thus should
+# be recovered by operators (i.e. disaster recovery procedure).
+# Unreliable networks should set this to a higher value.
+DEFAULT_ELECTION_FACTOR = 1
+
+
+class HealthState(Enum):
+    stable = auto()  # Network can commit new transactions
+    unavailable = auto()  # The primary or majority of nodes are unreachable
+    election = auto()  # An election is in progress
+
+
+def get_primary(node, client_node_timeout_s=3, verbose=True):
+    """
+    Returns the primary reported by a given node, and in which view or
+    None if the given node is unreachable.
+    """
+    try:
+        with node.client() as c:
+            logs = None if verbose else []
+            r = c.get(
+                "/node/network", timeout=client_node_timeout_s, log_capture=logs
```

Comment:
This will become `/node/consensus` once https://github.com/microsoft/CCF/pull/3666 is merged, so we also get the Raft state, and can give more precise state on the service health.